crates/providers/src/provider.rs

```diff
@@ -663,6 +673,26 @@ impl<T: Transport + Clone + Send + Sync> TempProvider for Provider<T> {
         self.inner.prepare("trace_block", block).await
     }
 
+    /// Sends a raw request with the methods and params specified to the internal connection,
+    /// and returns the result.
+    async fn raw_request<P, R>(
+        &self,
+        method: &'static str,
+        params: P,
+    ) -> TransportResult<R>
+    where
+        P: Serialize + Send + Sync + Clone,
+        R: Serialize + DeserializeOwned + Send + Sync + Unpin + 'static,
+        Self: Sync
+    {
+        let res = async move {
+            let res: R = self.inner.prepare(method, &params).await?;
+            Ok(res)
+        }.await;
+
+        res
```

Comment:
Why does this need a separate async block?

Reply:
yea no need, was experimenting with bounds for this. wil lremove